Library/List/SubscriptionAddOnList.cs

```diff
@@ -164,5 +164,20 @@ public void Add(string planAddOnCode, AddOn.Type addOnType, int quantity, int un
             var sub = new SubscriptionAddOn(planAddOnCode, addOnType, unitAmountInCents, quantity);
             base.Add(sub);
         }
+
+        // Supports item code as add-on code. Also supports plan add-on code as add-on code.
+        public void Add(string addOnCode, string addOnSource, int unitAmountInCents, int quantity = 1)
+        {
+          var sub = new SubscriptionAddOn(addOnCode, addOnSource, unitAmountInCents, quantity);
+          base.Add(sub);
+        }
+
+        public void Add(string planAddOnCode, string addOnSource, int quantity = 1)
+        {
+          var unitAmount = _subscription.Plan.AddOns.Find(ao => ao.AddOnCode == planAddOnCode).UnitAmountInCents[_subscription.Currency];
+          var sub = new SubscriptionAddOn(planAddOnCode, addOnSource, unitAmount, quantity);
+          base.Add(sub);
+        }
```

Comment:
Not sure if this is the best approach, but it was an attempt to be comprehensive and cover different scenarios.

The first `Add()` covers the the crux of items on subscriptions functionality. 

- It accepts an item code for the first param. 
- Second param `addOnSource` is required and must be "item". 
- Third param `unitAmountInCents` is also required when `addOnSource = item`. 

However, this function should also support plan add-on codes. The user may want to pass in "plan_add_on" for `addOnSource` and may want to change the `unitAmountInCents`. This function allows them to do that.

The second `Add()` allows the user to pass in a value for `addOnSource` while inheriting the `unitAmountInCents` from the plan add-on.